transformers/component_tests/flap/transformer_test.go

```diff
@@ -153,6 +156,45 @@ var _ = Describe("Executing the flap transformer", func() {
 		test_helpers.AssertVariable(liveResult, diff.ID, headerID, "1")
 	})
 
+	Describe("wards", func() {
+		It("reads in a wards storage diff row and persists it", func() {
+			denyLog := test_data.CreateTestLog(header.Id, db)
+			denyModel := test_data.DenyModel()
+
+			flapAddressID, flapAddressErr := shared.GetOrCreateAddress(test_data.FlapAddress(), db)
+			Expect(flapAddressErr).NotTo(HaveOccurred())
+
+			userAddress := "0x13141b8a5e4a82ebc6b636849dd6a515185d6236"
+			userAddressID, userAddressErr := shared.GetOrCreateAddress(userAddress, db)
+			Expect(userAddressErr).NotTo(HaveOccurred())
+
+			msgSenderAddress := "0x" + fakes.RandomString(40)
+			msgSenderAddressID, msgSenderAddressErr := shared.GetOrCreateAddress(msgSenderAddress, db)
+			Expect(msgSenderAddressErr).NotTo(HaveOccurred())
+
+			denyModel.ColumnValues[event.HeaderFK] = header.Id
+			denyModel.ColumnValues[event.LogFK] = denyLog.ID
+			denyModel.ColumnValues[event.AddressFK] = flapAddressID
+			denyModel.ColumnValues[constants.MsgSenderColumn] = msgSenderAddressID
+			denyModel.ColumnValues[constants.UsrColumn] = userAddressID
+			insertErr := event.PersistModels([]event.InsertionModel{denyModel}, db)
+			Expect(insertErr).NotTo(HaveOccurred())
+
+			key := common.HexToHash("614c9873ec2671d6eb30d7a22b531442a34fc10f8c24a6598ef401fe94d9cb7d")
```

Comment:
I wonder if we could go through the steps to calculate the key here so it's more explicit about where it came from?

Reply:
I like that idea, but in this instance I just grabbed a record from a CSV storage diff. Maybe it would be more thorough to actually calculate the key in the test though 🤔 

The code responsible for the calculation is pretty simple:
```
func GetKeyForMapping(indexOnContract, key string) common.Hash {
	keyBytes := common.FromHex(key + indexOnContract)
	return crypto.Keccak256Hash(keyBytes)
}
```